service/src/main/kotlin/no/nav/su/se/bakover/service/revurdering/RevurderingServiceImpl.kt

```diff
@@ -530,4 +576,132 @@ internal class RevurderingServiceImpl(
+        return Either.fromNullable(revurderingRepo.hent(request.revurderingId))
+            .mapLeft { FortsettEtterForhåndsvarselFeil.FantIkkeRevurdering }
+            .flatMap {
+                if (it !is SimulertRevurdering) {
+                    Either.left(FortsettEtterForhåndsvarselFeil.RevurderingErIkkeIRiktigTilstand)
+                } else {
+                    when (val forhåndsvarsel = it.forhåndsvarsel) {
+                        null ->
+                            Either.left(FortsettEtterForhåndsvarselFeil.RevurderingErIkkeForhåndsvarslet)
+                        is Forhåndsvarsel.SkalForhåndsvarsles.Besluttet ->
+                            Either.left(FortsettEtterForhåndsvarselFeil.AlleredeBesluttet)
+                        is Forhåndsvarsel.IngenForhåndsvarsel ->
+                            Either.left(FortsettEtterForhåndsvarselFeil.AlleredeBesluttet)
+                        is Forhåndsvarsel.SkalForhåndsvarsles.Sendt ->
+                            Either.right(Pair(it, forhåndsvarsel))
+                    }
+                }
+            }
+            .map { (revurdering, eksisterendeForhåndsvarsel) ->
+                val forhåndsvarsel = Forhåndsvarsel.SkalForhåndsvarsles.Besluttet(
+                    journalpostId = eksisterendeForhåndsvarsel.journalpostId,
+                    brevbestillingId = eksisterendeForhåndsvarsel.brevbestillingId,
+                    valg = utledBeslutningEtterForhåndsvarling(request),
+                    begrunnelse = request.begrunnelse,
+                )
+                revurderingRepo.oppdaterForhåndsvarsel(
+                    id = revurdering.id,
+                    forhåndsvarsel = forhåndsvarsel,
+                )
+
+                revurdering
+            }
+            .flatMap { revurdering ->
+                when (request) {
+                    is FortsettEtterForhåndsvarslingRequest.FortsettMedSammeOpplysninger -> {
+                        sendTilAttestering(
+                            SendTilAttesteringRequest(
+                                revurderingId = revurdering.id,
+                                saksbehandler = request.saksbehandler,
+                                fritekstTilBrev = request.fritekstTilBrev,
+                                skalFøreTilBrevutsending = true,
+                            ),
+                        ).mapLeft { FortsettEtterForhåndsvarselFeil.Attestering(it) }
+                    }
+                    is FortsettEtterForhåndsvarslingRequest.FortsettMedAndreOpplysninger -> {
+                        // Her er allerede revurderingen i riktig tilstand
+                        Either.right(revurdering)
+                    }
+                    is FortsettEtterForhåndsvarslingRequest.AvsluttUtenEndringer -> {
+                        TODO("Not yet implemented")
+                    }
+                }
+            }
+    }
+
+    private fun sendForhåndsvarsling(
```

Comment:
Har vi noe funksjonalitet for å få sendt ut brev hvis noe tryner her? Regner med at dette ikke glir rett inn i den funksjonaliteten vi har fra før for "fiksing av feil med brev".

Reply:
Feiler det, så feiler det. Må finne på noe her ja